cpp/arcticdb/storage/lmdb/lmdb_storage.cpp

```diff
@@ -316,7 +321,50 @@ LmdbStorage::LmdbStorage(const LibraryPath &library_path, OpenMode mode, const C
 
     txn.commit();
 
-    ARCTICDB_DEBUG(log::storage(), "Opened lmdb storage at {} with map size {}", lib_dir.generic_string(), format_bytes(mapsize));
+    ARCTICDB_DEBUG(log::storage(), "Opened lmdb storage at {} with map size {}", lib_dir_.string(), format_bytes(mapsize));
+}
+
+void LmdbStorage::warn_if_lmdb_already_open(const fs::path &root_path, const std::string &lib_path_str) {
+    uint64_t& count_for_pid = ++times_path_opened[(root_path / lib_path_str).string()];
+    if (count_for_pid != 1) {
+        // Strip magic name from warning as it will confuse users
+        if (lib_dir_.string().find(CONFIG_LIBRARY_NAME) == std::string::npos) {
+            log::storage().warn(fmt::format(
+                    "LMDB path at {} has already been opened in this process which is not supported by LMDB. "
+                    "To continue safely, you should delete all Library objects over the LMDB path in this "
+                    "process and then try again. This indicates a bug in ArcticDB. Please report at "
+                    "https://github.com/man-group/ArcticDB. Current process ID=[{}]",
+                    lib_dir_.string(), getpid()));
+        } else {
+            std::filesystem::path user_facing_path = lib_dir_;
+            user_facing_path.remove_filename();
+            log::storage().warn(fmt::format(
+                    "LMDB path at {} has already been opened in this process which is not supported by LMDB. "
+                    "You should only open a single Arctic instance over a given LMDB path. "
+                    "To continue safely, you should delete this Arctic instance and any others over the LMDB path in this "
+                    "process and then try again. Current process ID=[{}]",
+                    user_facing_path.string(), getpid()));
+        }
+    }
+}
+
+LmdbStorage::LmdbStorage(LmdbStorage&& other)  noexcept
+    : Storage(std::move(static_cast<Storage&>(other))),
+    write_mutex_(std::move(other.write_mutex_)),
+    env_(std::move(other.env_)),
+    dbi_by_key_type_(std::move(other.dbi_by_key_type_)),
+    lib_dir_(std::move(other.lib_dir_)) {
+    other.lib_dir_ = "";
+}
```

Comment:
Just noting that when Storages are made no move or copy in #991 I'll remove this move constructor. Is that going to be fine? It might mean the check of `!lib_dir_.empy()` in the destructor can go also, because `lib_dir` should always be passed something in the constructor.

Reply:
Proof of the pudding will be in the eating, but yeah that sounds good.